mewarpx/scripts_ancillary/tinyprofiler_plotting_example.py

```diff
@@ -0,0 +1,270 @@
+        for function in top["name"]:
+            function_names.add(function)
+
+    dataframes = []
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        print("plotting:", name)
+
+        f = open(name)
+        data = json.load(f)
+
+        gf = ht.GraphFrame.from_literal(data)
+        gf.drop_index_levels()
+
+        split_name = str(name).split("/")
+        cells = split_name[1]
+        cores = split_name[2].replace("core", "")
+
+        gf.dataframe = gf.dataframe.sort_values(by=parameter, ascending=False)
+
+        filtered_gf = gf.filter(lambda x : not math.isnan(x[parameter]) and x["name"] in function_names)
+
+        filtered_gf.dataframe["info"] = f"{cells}x{cells}\n{cores}"
+        if "gpu" not in cores:
+            filtered_gf.dataframe["info"] += "procs"
+
+        dataframes.append(filtered_gf.dataframe)
+
+    result = pd.concat(dataframes)
+
+    pivot_df = result.pivot(index="info", columns="name", values=parameter)
+
+    colors = []
+    # For this data I think tab20 was the best color choice
+    color = cm.tab20(np.linspace(0, 1, len(function_names)))
+    for i, c in zip(range(len(function_names)), color):
+        colors.append(c)
+
+    fig = pivot_df.loc[:,:].plot.bar(
+        stacked=False, figsize=(11,12),
+        title=f"Top {top_n} {description} times across regions",
+        ylabel=f"{description.capitalize()} time (s)", xlabel="Simulation Parameters",
+        logy=True, color=colors
+    )
+
+    fig.legend(bbox_to_anchor=(1,1))
+
+    plt.savefig(f"top_{top_n}_{description.replace(' ', '_')}_times.png", bbox_inches="tight")
+
+def plot_selection_of_times(functions_to_plot, exclusive, avg):
+    parameter = get_parameter_name(exclusive, avg)
+    description = get_description(exclusive, avg)
+
+    dataframes = []
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        print("plotting:", name)
+
+        f = open(name)
+        data = json.load(f)
+
+        gf = ht.GraphFrame.from_literal(data)
+        gf.drop_index_levels()
+
+        split_name = str(name).split("/")
+        cells = split_name[1]
+        cores = split_name[2].replace("core", "")
+
+        gf.dataframe = gf.dataframe.sort_values(by=parameter, ascending=False)
+
+        filtered_gf = gf.filter(lambda x : not math.isnan(x[parameter]) and x["name"] in functions_to_plot)
+
+        filtered_gf.dataframe["info"] = f"{cells}x{cells}\n{cores}"
+        if "gpu" not in cores:
+            filtered_gf.dataframe["info"] += "procs"
+
+        dataframes.append(filtered_gf.dataframe)
+
+    result = pd.concat(dataframes)
+
+    pivot_df = result.pivot(index="info", columns="name", values=parameter)
+
+    # plot the data
+    fig = pivot_df.loc[:,:].plot.bar(
+        stacked=False, figsize=(11,12),
+        title=f"Selected {description} function times across regions",
+        ylabel=f"{description.capitalize()} time (s)", xlabel="Simulation Parameters",
+        logy=True
+    )
```

Comment:
Consider more separate functions / code re-use here?

In fact can the `plot_union_of_times` just call this function once it determines what functions to plot, with a specified colormap and a specified title/etc.?

Reply:
Yep that'll be better